tests/rule_based_profiler/parameter_builder/test_parameter_builder.py

```diff
@@ -0,0 +1,486 @@
+        )
+
+        self._call_count: int = 0
+
+    @property
+    def call_count(self) -> int:
+        return self._call_count
+
+    def _build_parameters(
+        self,
+        domain: Domain,
+        variables: Optional[ParameterContainer] = None,
+        parameters: Optional[Dict[str, ParameterContainer]] = None,
+        runtime_configuration: Optional[dict] = None,
+    ) -> Attributes:
+        """
+        Only "_call_count" is incremented; otherwise, arbitrary test value is returned.
+        """
+        self._call_count += 1
+
+        return Attributes(
+            {
+                FULLY_QUALIFIED_PARAMETER_NAME_VALUE_KEY: 1,
+                FULLY_QUALIFIED_PARAMETER_NAME_ATTRIBUTED_VALUE_KEY: {"batch_id_0": 1},
+                FULLY_QUALIFIED_PARAMETER_NAME_METADATA_KEY: {"environment": "test"},
+            }
+        )
+
+
+# noinspection PyUnresolvedReferences,PyUnusedLocal
+@pytest.mark.unit
+@mock.patch(
+    "great_expectations.rule_based_profiler.parameter_builder.parameter_builder.convert_to_json_serializable",
+    return_value="my_json_string",
+)
+def test_parameter_builder_should_not_recompute_evaluation_parameter_builders_if_precomputed(
+    mock_convert_to_json_serializable: mock.MagicMock,
+):
+    my_evaluation_dependency_0_parameter_builder: ParameterBuilder = (
+        DummyParameterBuilder(
+            name="my_evaluation_dependency_parameter_name_0",
+            evaluation_parameter_builder_configs=None,
+        )
+    )
+    my_evaluation_dependency_1_parameter_builder: ParameterBuilder = (
+        DummyParameterBuilder(
+            name="my_evaluation_dependency_parameter_name_1",
+            evaluation_parameter_builder_configs=None,
+        )
+    )
+
+    my_dependent_parameter_builder: ParameterBuilder = DummyParameterBuilder(
+        name="my_dependent_parameter_builder",
+        evaluation_parameter_builder_configs=None,
+    )
+
+    domain = DummyDomain()
+
+    parameter_container = ParameterContainer(parameter_nodes=None)
+    parameters: Dict[str, ParameterContainer] = {
+        domain.id: parameter_container,
+    }
+
+    my_evaluation_dependency_0_parameter_builder.build_parameters(
+        domain=domain,
+        variables=None,
+        parameters=parameters,
+        batch_request=None,
+        runtime_configuration=None,
+    )
+    assert my_evaluation_dependency_0_parameter_builder.call_count == 1
+
+    my_evaluation_dependency_1_parameter_builder.build_parameters(
+        domain=domain,
+        variables=None,
+        parameters=parameters,
+        batch_request=None,
+        runtime_configuration=None,
+    )
+    assert my_evaluation_dependency_1_parameter_builder.call_count == 1
+
+    dependencies_fully_qualified_parameter_names: List[
+        str
+    ] = get_fully_qualified_parameter_names(
+        domain=domain,
+        variables=None,
+        parameters=parameters,
+    )
+
+    my_dependent_parameter_builder.resolve_evaluation_dependencies(
```

Comment:
what does this do and why is it needed as part of this test's setup? Specifically, there are assertions below and I'm not sure how or if they relate to this method, and there are no assertions here to either explain or validate the behavior of this method

Reply:
@donaldheppner This tests one of the behaviors: resolution of prerequisite ParameterBuilders.  The need for the setup stems from the fact that all ParameterBuilder computations output result into shared memory.  Resolution of prerequisite ParameterBuilders (via `ParameterBuilder.resolve_evaluation_dependencies()`) involves ParameterBuilder computation -- it runs `ParameterBuilder.build_parameters()` for all prerequisite ParameterBuilders, thereby all updating the same shared memory space.  Hence, this set up is for the behavior that is key part of the overall ParameterBuilder behavior.  It is tested as part of the `ParameterBuilder.build_parameters()` because it is part of it, and itself involves execution of `ParameterBuilder.build_parameters()`.  The assertions are as follows: On line 178, we assert that the progress of filling the shared memory from resolving the prerequisite ParameterBuilders is as expected.  In the next test, we do the same (but for a different expected result) on line 293.  I have added another docstring in order to help making this more clear to the reader.  Please let me know if this helps.  Thank you.